### PR TITLE
Move published modules to standalone plugins

### LINKED ISSUE #654: Remove the modules that were published as standalone plugins from the Performance Lab plugin
## Feature Description

Follow up to #651 #652 #653: The final part of this workstream will be to actually remove the modules that are published as standalone plugins from the PL plugin. This will not actually involve deleting the folders in the repository, as they will still be developed here (i.e. as a monorepo), instead we will need to ensure they are excluded from the plugin, maybe move to a new `plugins` directory in this repo, and at that point we can also change their file headers to actual plugin headers instead of PL module headers.

### Requirements

**This work should be implemented against a new `feature/modules-to-plugins` feature branch.**

After #934 has been completed:
* Move the folders for the modules referenced in `plugins.json` into the new `plugins` folder (without their focus area folders).
* Keep those modules/plugins unchanged, _except_ for:
    * Update the respective `load.php` file to have the plugin header instead of the module header.
    * The plugin header should have the same content as if it had been built using our current "module to plugin" build workflow.
* Migrate the existing `plugins.json` entries from the `modules` property to the `plugins` property.
* Update the references to the migrated modules in `CODEOWNERS` to point to the respective plugin folders.
* Remove references to the migrated modules in the `perflab_get_standalone_plugins_constants()` function.

### PR DESCRIPTION
**Pull request by thelovekesh**
## Summary

- Move published modules to the `plugins` directory. Moved modules include:
    - `dominant-color-images`
    - `webp-uploads`
- Move modules test cases to `tests/plugins` directory.
- Add testsuite entry in the PHPUnit config.
- Update plugin files headers and doc blocks.
- Update `@since` and `@package` tags.

Fixes #654 

## Relevant technical choices

- Update `@since` tags based on the module changelog.
- Update `@package` tags based on the module name.
- Rename the main plugin file based on the plugin name.
- Remove checks included to return if "module" was already activated.
- Update plugin version as per `plugin.json` metadata.

## Checklist

- [x] PR has either `[Focus]` or `Infrastructure` label.
- [x] PR has a `[Type]` label.
- [x] PR has a milestone or the `no milestone` label.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -49,11 +44,15 @@
...
-/tests/modules/images/dominant-color-images                                 @pbearne @spacedmonkey
-/tests/testdata/modules/images/dominant-color-images                        @pbearne @spacedmonkey
-
 # Plugin: Auto-sizes for Lazy-Loaded Images
 /plugins/auto-sizes                         @joemcgill
 /tests/plugins/auto-sizes                   @joemcgill
+
+# Plugin: WebP Uploads
+/plugins/webp-uploads                       @adamsilverstein @felixarntz
+/tests/plugins/webp-uploads                 @adamsilverstein @felixarntz
```
Comment (westonruter) [inline .github/CODEOWNERS:57]: What about `/tests/testdata/plugins/webp-uploads`? This was present above but it is missing here.

File: plugins.json [hunk 2]
```diff
@@ -9,5 +9,9 @@
       "version": "1.0.5"
     }
   },
-  "plugins": [ "auto-sizes" ]
+  "plugins": [
+    "auto-sizes",
+    "dominant-color-images",
+    "webp-uploads"
```
Comment (westonruter) [inline plugins.json]: Versions will need to be added once https://github.com/WordPress/performance/pull/1000 is merged.

File: plugins/dominant-color-images/.gitattributes
Comment (westonruter) [inline plugins/dominant-color-images/.gitattributes:1]: Shouldn't this `.gitattributes` file rather be deleted in favor of having this information in the root `.gitattributes`?

File: plugins/dominant-color-images/class-dominant-color-image-editor-imagick.php
Comment (westonruter) [inline plugins/dominant-color-images/class-dominant-color-image-editor-imagick.php:1]: The text domains in this file (and maybe others) need to be updated from `performance-lab` to `dominant-color-images`. See https://plugins.trac.wordpress.org/browser/dominant-color-images/tags/1.0.1/class-dominant-color-image-editor-imagick.php#L64

File: plugins/webp-uploads/.gitattributes
Comment (westonruter) [inline plugins/webp-uploads/.gitattributes:1]: Ditto from above regarding this file.

File: plugins/webp-uploads/helper.php [hunk 3]
```diff
@@ -2,14 +2,11 @@
...
- * @package performance-lab
+ * @package webp-uploads
+ *
  * @since 1.0.0
  */
 
-if ( ! defined( 'ABSPATH' ) ) {
-	exit; // Exit if accessed directly.
-}
-
```
Comment (westonruter) [inline plugins/webp-uploads/helper.php]: Why is this deleted?

**Review by westonruter — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (thelovekesh) [reply .github/CODEOWNERS:57]: Moved it to `tests/plugins/webp-uploads/helpers` for being autoloaded by PHPUnit in https://github.com/WordPress/performance/pull/1011/commits/370a592e4fe2c54f65c05e0fed12d5ebb0fe9868

File: plugins/dominant-color-images/.gitattributes
Comment (thelovekesh) [reply plugins/dominant-color-images/.gitattributes:1]: Yes, makes sense to me. Also, I don't see its benefit in the current setup since it's being added in the dist plugin.

File: plugins/dominant-color-images/class-dominant-color-image-editor-imagick.php
Comment (thelovekesh) [reply plugins/dominant-color-images/class-dominant-color-image-editor-imagick.php:1]: Aah yes. I missed it. Thanks for pointing out.

File: plugins/webp-uploads/helper.php (hunk 3 above)
Comment (thelovekesh) [reply plugins/webp-uploads/helper.php]: I am not sure if I missed any use cases. Since this file is just declaring functions and not invoking them, do we still need this safeguard?

File: phpcs.xml.dist [hunk 4]
```diff
@@ -8,7 +8,7 @@
 	<rule ref="WordPress-Docs"/>
 	<rule ref="WordPress-Extra" />
 	<rule ref="WordPress.WP.I18n"/>
-	<config name="text_domain" value="performance-lab,default,auto-sizes"/>
+	<config name="text_domain" value="performance-lab,default,auto-sizes,dominant-color-images,webp-uploads"/>
```
Comment (thelovekesh) [inline phpcs.xml.dist]: Note: Needs to be updated once https://github.com/WordPress/performance/pull/1002 is merged.

**Comment by thelovekesh** [comment]
@mukeshpanchal27 JFYI recently added unit tests setup for standalone plugins doesn't seem to be running multisite test cases - https://github.com/WordPress/performance/actions/runs/8027561041/job/21931681466?pr=1011. I am aiming to amend this setup in https://github.com/WordPress/performance/pull/1013.

File: plugins/webp-uploads/helper.php (hunk 3 above)
Comment (mukeshpanchal27) [reply plugins/webp-uploads/helper.php]: Yes.

**Comment by mukeshpanchal27** [comment]
> @mukeshpanchal27 JFYI recently added unit tests setup for standalone plugins doesn't seem to be running multisite test cases - https://github.com/WordPress/performance/actions/runs/8027561041/job/21931681466?pr=1011. I am aiming to amend this setup in #1013.

Great catch. It was indeed missed. I've added the missing configuration in PR #1014.

File: default-enabled-modules.php [hunk 5]
```diff
@@ -1,8 +1,6 @@
 <?php
 /* THIS IS A GENERATED FILE. DO NOT EDIT DIRECTLY. */
 return array(
-	'images/dominant-color-images',
```
Comment (mukeshpanchal27) [inline default-enabled-modules.php]: Note: It will auto updated in release process. Check https://make.wordpress.org/performance/handbook/performance-lab/releasing-the-plugin/#update-list-of-default-enabled-modules

File: plugins/dominant-color-images/dominant-color-images.php [hunk 6]
```diff
@@ -0,0 +1,27 @@
+<?php
```
Comment (mukeshpanchal27) [inline plugins/dominant-color-images/dominant-color-images.php]: The plugin released with [load.php](https://plugins.trac.wordpress.org/browser/dominant-color-images/trunk/load.php) we should rename this file back to `load.php`

File: plugins/webp-uploads/webp-uploads.php [hunk 7]
```diff
@@ -0,0 +1,30 @@
+<?php
```
Comment (mukeshpanchal27) [inline plugins/webp-uploads/webp-uploads.php]: Check similar comment `dominant-color-images` plugin.

File: plugins/dominant-color-images/dominant-color-images.php [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+/**
+ * Plugin Name: Dominant Color Images
+ * Plugin URI: https://github.com/WordPress/performance/tree/trunk/modules/images/dominant-color-images
```
Comment (mukeshpanchal27) [inline plugins/dominant-color-images/dominant-color-images.php]: This should links to `/plugins/` folder now.

File: plugins/webp-uploads/webp-uploads.php [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * Plugin Name: WebP Uploads
+ * Plugin URI: https://github.com/WordPress/performance/tree/trunk/modules/images/webp-uploads
```
Comment (mukeshpanchal27) [inline plugins/webp-uploads/webp-uploads.php]: This should links to `/plugins/` folder now.

File: .github/CODEOWNERS [hunk 10]
```diff
@@ -49,11 +44,15 @@
...
+
+# Plugin: Dominant Color Images
+/plugins/dominant-color-images                                              @pbearne @spacedmonkey
+/tests/plugins/dominant-color-images                                        @pbearne @spacedmonkey
+/tests/testdata/modules/images/dominant-color-images                        @pbearne @spacedmonkey
```
Comment (mukeshpanchal27) [inline .github/CODEOWNERS]: Per https://github.com/WordPress/performance/pull/1011/files#r1501248677 do we need to move this into helpers?

**Review by mukeshpanchal27 — CHANGES_REQUESTED** [review]
Thanks @thelovekesh for the PR. This PR is blocked by #1000 and #1002. Left some initial feedbacks.

File: .github/CODEOWNERS (hunk 10 above)
Comment (thelovekesh) [reply .github/CODEOWNERS]: Hmm, yes, because it would be better to keep things related to a standalone plugin in a single place.

File: plugins/webp-uploads/helper.php (hunk 3 above)
Comment (thelovekesh) [reply plugins/webp-uploads/helper.php]: @mukeshpanchal27 Not sure I understand. Can you please share some info here?

File: .github/CODEOWNERS (hunk 10 above)
Comment (thelovekesh) [reply .github/CODEOWNERS]: done in https://github.com/WordPress/performance/pull/1011/commits/69b6275b08602d0b7fff6e5f7886a145c17a700a

File: plugins/webp-uploads/helper.php (hunk 3 above)
Comment (westonruter) [reply plugins/webp-uploads/helper.php]: As I recall, the plugin review team requested this short-circuit logic. But I don't think it is in the handbook, so it may not be a hard requirement. I think @joemcgill got that feedback.

**Review by westonruter — APPROVED** [review]
LGTM

**Review by felixarntz — CHANGES_REQUESTED** [review]
@thelovekesh This looks good so far.

I'm putting a "Request changes" for now, just so that we don't merge it yet until the foundation has been completed. I'm worried that if we merge this already while the foundation is still being worked on in other PRs (e.g. #1002 that you highlighted) we'll run into merge conflicts and things may become too confusing. Let's merge the removal/migration of the modules as the last step.

**Comment by thelovekesh** [comment]
@felixarntz @mukeshpanchal27 Since #1000 and #1002 are now merged, we can pick it up next?

**Comment by mukeshpanchal27** [comment]
@thelovekesh, could you please rebase the PR as it seems to have conflicts?

**Comment by thelovekesh** [comment]
@mukeshpanchal27 Resolved the conflicts, Can you please take a look now?

File: plugins/dominant-color-images/load.php [hunk 11]
```diff
@@ -0,0 +1,27 @@
+<?php
+/**
+ * Plugin Name: Dominant Color Images
+ * Plugin URI: https://github.com/WordPress/performance/tree/trunk/plugins/images/dominant-color-images
```
Comment (mukeshpanchal27) [inline plugins/dominant-color-images/load.php]:
```suggestion
 * Plugin URI: https://github.com/WordPress/performance/tree/trunk/plugins/dominant-color-images
```

File: plugins/webp-uploads/load.php [hunk 12]
```diff
@@ -0,0 +1,30 @@
+<?php
+/**
+ * Plugin Name: WebP Uploads
+ * Plugin URI: https://github.com/WordPress/performance/tree/trunk/plugins/images/webp-uploads
```
Comment (mukeshpanchal27) [inline plugins/webp-uploads/load.php]:
```suggestion
 * Plugin URI: https://github.com/WordPress/performance/tree/trunk/plugins/webp-uploads
```

File: plugins/dominant-color-images/load.php [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+// Define required constants.
+define( 'DOMINANT_COLOR_IMAGES_VERSION', '1.0.1' );
+
```
Comment (mukeshpanchal27) [inline plugins/dominant-color-images/load.php:29]: Why removed below function from file? Without this code it generate `Plugin could not be activated because it triggered a fatal error.` when some one use old PL plugin and new module.

```
// Do not load the code if it is already loaded through another means.
if ( function_exists( 'dominant_color_metadata' ) ) {
	return;
}
```

File: plugins/webp-uploads/load.php [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+// Define required constants.
+define( 'WEBP_UPLOADS_VERSION', '1.0.5' );
+
```
Comment (mukeshpanchal27) [inline plugins/webp-uploads/load.php:29]: Why removed below function from file? Without this code it generate `Plugin could not be activated because it triggered a fatal error.` when some one use old PL plugin and new module.

```
// Do not load the code if it is already loaded through another means.
if ( function_exists( 'webp_uploads_create_sources_property' ) ) {
	return;
}
```

File: plugins/webp-uploads/helper.php (hunk 3 above)
Comment (mukeshpanchal27) [reply plugins/webp-uploads/helper.php]: Check #945 for more details.

**Review by mukeshpanchal27 — DISMISSED** [review]

File: plugins.json [hunk 15]
```diff
@@ -9,5 +9,9 @@
       "version": "1.0.5"
     }
   },
```
Comment (mukeshpanchal27) [inline plugins.json]: Remove the modules.

```
"modules": {}
```

File: plugins/dominant-color-images/load.php (hunk 13 above)
Comment (thelovekesh) [reply plugins/dominant-color-images/load.php:29]: Shouldn't the plugin take precedence over the module if it's published as a plugin?

Comment (westonruter) [reply plugins/dominant-color-images/load.php:29]: Yeah, that makes sense to me. Old PL versions with that `if` intact will short circuit when the standalone plugins is active.

**Comment by felixarntz** [comment]
Please take a look at https://github.com/WordPress/performance/issues/656#issuecomment-1969542264 where we can coordinate the next steps on how to proceed with this PR and #1019.

**Comment by thelovekesh** [comment]
@felixarntz I think we can change the base to trunk now?

**Comment by felixarntz** [comment]
@thelovekesh Thanks, I've updated it. There are a few small merge conflicts, but they should be straightforward to resolve.

File: .github/CODEOWNERS [hunk 16]
```diff
@@ -49,15 +44,18 @@
...
 /plugins/speculation-rules                  @felixarntz
 /tests/plugins/speculation-rules            @felixarntz
+
+# Plugin: WebP Uploads
+/plugins/webp-uploads                       @adamsilverstein @felixarntz
+/tests/plugins/webp-uploads                 @adamsilverstein @felixarntz
+
+# Plugin: Dominant Color Images
+/plugins/dominant-color-images                                              @pbearne @spacedmonkey
+/tests/plugins/dominant-color-images                                        @pbearne @spacedmonkey
```
Comment (felixarntz) [inline .github/CODEOWNERS]: Nit-pick, let's align the spacing with the other entries.

File: phpunit.xml.dist [hunk 17]
```diff
@@ -18,6 +18,12 @@
 		<testsuite name="speculation-rules">
 			<directory suffix=".php">./tests/plugins/speculation-rules</directory>
 		</testsuite>
+		<testsuite name="dominant-color-images">
+			<directory suffix=".php">./tests/plugins/dominant-color-images</directory>
+		</testsuite>
+		<testsuite name="webp-uploads">
+			<directory suffix=".php">./tests/plugins/webp-uploads</directory>
+		</testsuite>
```
Comment (felixarntz) [inline phpunit.xml.dist]: Nit-pick: In e.g. `composer.json` and `plugins.json`, you sort the plugins alphabetically. Maybe we can do that here too?
```suggestion
		<testsuite name="dominant-color-images">
			<directory suffix=".php">./tests/plugins/dominant-color-images</directory>
		</testsuite>
		<testsuite name="speculation-rules">
			<directory suffix=".php">./tests/plugins/speculation-rules</directory>
		</testsuite>
		<testsuite name="webp-uploads">
			<directory suffix=".php">./tests/plugins/webp-uploads</directory>
		</testsuite>
```

File: plugins/dominant-color-images/load.php (hunk 13 above)
Comment (felixarntz) [reply plugins/dominant-color-images/load.php:29]: It's safer to keep the `defined( ... )` constant check above the constant to prevent double loading. Yes, this probably shouldn't happen, but IMO better safe than sorry, and the other standalone plugins have that condition too.

If we decide to remove them, let's remove them everywhere rather than making partial decisions.

File: plugins/dominant-color-images/load.php [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * @package dominant-color-images
+ */
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+// Define required constants.
```
Comment (felixarntz) [inline plugins/dominant-color-images/load.php:23]: See other comment thread:
```suggestion
if ( defined( 'DOMINANT_COLOR_IMAGES_VERSION' ) ) {
	return;
}

```

File: plugins/webp-uploads/load.php (hunk 14 above)
Comment (felixarntz) [reply plugins/webp-uploads/load.php:29]: The function check shouldn't be needed anymore, it was introduced before the versions constants existed. But for safety and consistency, we should keep the constant check in place for now (see my other comment).

File: plugins/webp-uploads/load.php [hunk 19]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ * @package webp-uploads
+ */
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+// Define required constants.
```
Comment (felixarntz) [inline plugins/webp-uploads/load.php:23]: See other comment:
```suggestion
if ( defined( 'WEBP_UPLOADS_VERSION' ) ) {
	return;
}
```

File: tests/plugins/webp-uploads/helpers/class-wp-image-doesnt-support-webp.php [hunk 20]
```diff
@@ -2,7 +2,7 @@
 /**
  * A WP_Image_Editor mock that doesn't support WebP.
```
Comment (felixarntz) [inline tests/plugins/webp-uploads/helpers/class-wp-image-doesnt-support-webp.php]: Let's name this folder `data` instead of `helpers`, similar to how you did it for Dominant Color Images, and in accordance with both the original files being in the `testdata` folder.

File: tests/plugins/webp-uploads/helpers/class-wp-image-edit.php [hunk 21]
```diff
@@ -2,7 +2,7 @@
 /**
  * A WP_Image_Edit class to avoid abstraction to handle image edits.
  *
- * @package performance-lab
+ * @package webp-uploads
```
Comment (felixarntz) [inline tests/plugins/webp-uploads/helpers/class-wp-image-edit.php]: See above, the folder with this file should have a consistent name with how you did it for Dominant Color Images.

**Review by felixarntz — COMMENTED** [review]
@thelovekesh This is almost good to go, only some minor feedback.

One overall concern: It looks like you removed all the individual `.gitattributes` files. Don't we still need them to ensure the extra files aren't part of the plugin ZIP builds? We will at least need to exclude `.wordpress-org` and `phpcs.xml.dist` for every plugin.

**Review by mukeshpanchal27 — COMMENTED** [review]
For consistency, could we use the `data` folder within the `webp-uploads` plugin to store images for testing, similar to how it's done for the `dominant-color-images` plugin? Currently, `webp-uploads` uses images from the `tests/testdata/modules/images` folder.

File: phpunit.xml.dist (hunk 17 above)
Comment (thelovekesh) [reply phpunit.xml.dist]: Seems like I missed it while resolving merge conflicts.

File: .github/CODEOWNERS (hunk 16 above)
Comment (thelovekesh) [reply .github/CODEOWNERS]: Done https://github.com/WordPress/performance/pull/1011/commits/486014e1b6bd074ebbf110ef6a234b2248178c97

File: plugins/dominant-color-images/load.php (hunk 18 above)
Comment (thelovekesh) [reply plugins/dominant-color-images/load.php:23]: Done in https://github.com/WordPress/performance/pull/1011/commits/a6c1ddb7f219e09faaec4a396f7c9e78de90bdc5

File: plugins/webp-uploads/load.php (hunk 19 above)
Comment (thelovekesh) [reply plugins/webp-uploads/load.php:23]: Done in https://github.com/WordPress/performance/pull/1011/commits/a6c1ddb7f219e09faaec4a396f7c9e78de90bdc5

File: plugins/dominant-color-images/load.php (hunk 13 above)
Comment (thelovekesh) [reply plugins/dominant-color-images/load.php:29]: done in https://github.com/WordPress/performance/pull/1011/commits/a6c1ddb7f219e09faaec4a396f7c9e78de90bdc5

File: plugins/webp-uploads/load.php (hunk 14 above)
Comment (thelovekesh) [reply plugins/webp-uploads/load.php:29]: done in https://github.com/WordPress/performance/pull/1011/commits/a6c1ddb7f219e09faaec4a396f7c9e78de90bdc5

File: tests/plugins/webp-uploads/helpers/class-wp-image-doesnt-support-webp.php (hunk 20 above)
Comment (thelovekesh) [reply tests/plugins/webp-uploads/helpers/class-wp-image-doesnt-support-webp.php]: done in https://github.com/WordPress/performance/pull/1011/commits/de7732047571df69397865caf202c80a39331475

File: tests/plugins/webp-uploads/helpers/class-wp-image-edit.php (hunk 21 above)
Comment (thelovekesh) [reply tests/plugins/webp-uploads/helpers/class-wp-image-edit.php]: Renamed in https://github.com/WordPress/performance/pull/1011/commits/de7732047571df69397865caf202c80a39331475

**Comment by thelovekesh** [comment]
> For consistency, could we use the `data` folder within the `webp-uploads` plugin to store images for testing, similar to how it's done for the `dominant-color-images` plugin? Currently, `webp-uploads` uses images from the `tests/testdata/modules/images` folder.

@mukeshpanchal27 I think those images are used by `dominant-color-images` plugin as well. IMO we can keep shared resources in testdata and standalone in plugin-specific data directory?

**Comment by thelovekesh** [comment]
> One overall concern: It looks like you removed all the individual .gitattributes files. Don't we still need them to ensure the extra files aren't part of the plugin ZIP builds? We will at least need to exclude .wordpress-org and phpcs.xml.dist for every plugin.

@felixarntz I removed it since it looks like it's not being used. Since `.gitattributes` is git specific, I can't find any instances in the build/deploy setup where this is being used for standalone plugins or modules. As a result, I can see ignored files/directories are there in the published plugin as well - https://plugins.svn.wordpress.org/dominant-color-images/trunk/.

Also, we might need to switch from `.gitattributes` to `.distignore`(see https://github.com/WordPress/performance/issues/893#issuecomment-1947398713) since some WIP plugins have assets that are not in the git tree but need to be included in the plugin release.

**Comment by felixarntz** [comment]
@thelovekesh `.gitattributes` is supported by the 10up build action, no need to mention it in any of our build/deploy workflows. That's why it's used in the root of the project for the main PL plugin as well. At some point in the past we used `.distignore`, but it was less practical as it's not considered by GitHub itself, as far as I can remember.

Let me take a look at the `dominant-color-images` plugin, I believe the assets are in that one because of the same bug happening in the past, and they would need to be manually deleted to no longer be in there.

Let's use `.gitattributes` for now, as we have been, and we can rethink this in the future.

**Comment by thelovekesh** [comment]
@felixarntz I think the 10up deploy action only takes `.gitattributes` into account if [`BUILD_DIR` is false](https://github.com/10up/action-wordpress-plugin-deploy/blob/abb939a0d0bfd01063e8d1933833209201557381/deploy.sh#L92). For the main plugin we don't pass any `BUILD_DIR` but for the standalone plugins we do. Can you please validate it once?

**Comment by felixarntz** [comment]
@thelovekesh Oh, that's a great catch! So it looks like there's no way to use either of the two files for the 10up deploy workflow to automatically handle it?

Maybe we have to implement something in our own workflow that removes those files before triggering the 10up deploy workflow? 🤔

**Comment by thelovekesh** [comment]
@felixarntz Yes, and I think instead of handling it with manual CLI script we should switch to a build tool to better handle bundling use cases.

As far as we are concerned about `.distignore`, it shouldn't hurt our workflow in any way. `.gitattributes` play nice when we only want to build the stuff that is in the git tree.

---

Coming to the release of standalone plugins, we need to move WPOrg assets to some place(maybe tmp?) and then pass it path in `ASSETS_DIR`. It should do the trick.

**Comment by felixarntz** [comment]
@thelovekesh In the interest of unblocking other PRs, let's merge this PR now without the correct handling of the assets.

Similarly to what I outlined in #1008, it would be great if you could open another PR against `trunk` to fix specifically that bug, to ensure that development files and brand assets for the standalone plugins are not part of the builds and deployments.

**Review by felixarntz — APPROVED** [review]
